core/db/src/main/java/org/opennms/core/db/DataSourceFactory.java

```diff
@@ -205,7 +206,10 @@ public static void setInstance(final DataSource ds) {
      * @param ds a {@link javax.sql.DataSource} object.
      */
     public static synchronized void setInstance(final String dsName, final DataSource ds) {
-        m_dataSources.put(dsName, ds);
+        DataSource oldDs = m_dataSources.put(dsName, ds);
+        if (oldDs != null) {
+            // TODO: Call the closer Runnable?
```

Comment:
The TODO just tags this line as a possible problem that needs further investigation.